transformations/underscore_trick/transformation.py

```diff
@@ -0,0 +1,42 @@
+import itertools
+import random
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+def add_underscore(text, prob=0.05, seed=42, max_outputs=1):
+    random.seed(seed)
+
+    perturbed_texts = []
+    for _ in itertools.repeat(None, max_outputs):
+        perturbed_text = "".join(
+            [
+                letter if letter != " " or random.random() > prob else "_"
+                for letter in text
+            ]
+        )
+        perturbed_texts.append(perturbed_text)
+    return perturbed_texts
+
+
+class UnderscoreTrick(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["All"]
+
+    def __init__(self, seed=42, max_outputs=1, prob=0.05):
```

Comment:
Can you pass a variable here which takes a character as input? This could be expanded to other characters too like "-", "@", "#", etc. Might not be a good idea to create a separate transformation for each.

Reply:
Thank you @kaustubhdhole for the suggestion. I have now implemented it in 9b2fc383d10c8bf5e2f9c6a93e8eac8a2dc3c759